packages/hdwallet-xdefi/src/xdefi.ts

```diff
@@ -0,0 +1,340 @@
+  }
+
+  public getVendor(): string {
+    return "XDeFi";
+  }
+
+  public getModel(): Promise<string> {
+    return Promise.resolve("XDeFi");
+  }
+
+  public getLabel(): Promise<string> {
+    return Promise.resolve("XDeFi");
+  }
+
+  public async initialize(): Promise<any> {
+    try {
+      this.provider = (window as any).xfi && (window as any).xfi.ethereum;
+    } catch (e) {
+      console.error(e);
+    }
+
+    return Promise.resolve();
+  }
+
+  public hasOnDevicePinEntry(): boolean {
+    return this.info.hasOnDevicePinEntry();
+  }
+
+  public hasOnDevicePassphrase(): boolean {
+    return this.info.hasOnDevicePassphrase();
+  }
+
+  public hasOnDeviceDisplay(): boolean {
+    return this.info.hasOnDeviceDisplay();
+  }
+
+  public hasOnDeviceRecovery(): boolean {
+    return this.info.hasOnDeviceRecovery();
+  }
+
+  public hasNativeShapeShift(srcCoin: core.Coin, dstCoin: core.Coin): boolean {
+    return this.info.hasNativeShapeShift(srcCoin, dstCoin);
+  }
+
+  public supportsOfflineSigning(): boolean {
+    return false;
+  }
+
+  public supportsBroadcast(): boolean {
+    return true;
+  }
+
+  public async clearSession(): Promise<void> {
+    // TODO: Can we lock XDeFi from here?
+  }
+
+  public ping(msg: core.Ping): Promise<core.Pong> {
+    // no ping function for XDeFi, so just returning Core.Pong
+    return Promise.resolve({ msg: msg.msg });
+  }
+
+  public sendPin(pin: string): Promise<void> {
+    // no concept of pin in XDeFi
+    return Promise.resolve();
+  }
+
+  public sendPassphrase(passphrase: string): Promise<void> {
+    // cannot send passphrase to XDeFi. Could show the widget?
+    return Promise.resolve();
+  }
+
+  public sendCharacter(charater: string): Promise<void> {
+    // no concept of sendCharacter in XDeFi
+    return Promise.resolve();
+  }
+
+  public sendWord(word: string): Promise<void> {
+    // no concept of sendWord in XDeFi
+    return Promise.resolve();
+  }
+
+  public cancel(): Promise<void> {
+    // no concept of cancel in XDeFi
+    return Promise.resolve();
+  }
+
+  public wipe(): Promise<void> {
+    return Promise.resolve();
+  }
+
+  public reset(msg: core.ResetDevice): Promise<void> {
+    return Promise.resolve();
+  }
+
+  public recover(msg: core.RecoverDevice): Promise<void> {
+    // no concept of recover in XDeFi
+    return Promise.resolve();
+  }
+
+  public loadDevice(msg: core.LoadDevice): Promise<void> {
+    // TODO: Does XDeFi allow this to be done programatically?
+    return Promise.resolve();
+  }
+
+  public describePath(msg: core.DescribePath): core.PathDescription {
+    return this.info.describePath(msg);
+  }
+
+  public async getPublicKeys(msg: Array<core.GetPublicKey>): Promise<Array<core.PublicKey | null>> {
+    // Ethereum public keys are not exposed by the RPC API
+    return [];
+  }
+
+  public async isInitialized(): Promise<boolean> {
+    return true;
+  }
+
+  public disconnect(): Promise<void> {
+    return Promise.resolve();
+  }
```

Comment:
```suggestion
  public async disconnect(): Promise<void> { }
```

Reply:
This needs a promise to be resolved as that's how the signature is in the interface `core.HDWallet`, otherwise Typescript complains.

Reply:
`async` guarantees a `Promise` is returned so I would think this should be fine.

Reply:
That is true, but we aren't returning anything in the suggestion right? Typescript complains that a value must be returned 
`A function whose declared type is neither 'void' nor 'any' must return a value.ts(2355)`

Reply:
Please ignore the previous comment as I missed out on the `async` bit.